Added documentation to lib_camera_v2 (#119)
xmos/lib_camera: doc/rst/07_api_reference.rst

Reviewer: :noindex: please

Reviewer: I would probably restructure this header by relevance to the user:
camera main
io
isp
sensors

Reviewer: each of the subheadings needs a sentence or 2, otherwise, it looks like it's written for robots

|newpage|

.. _lib_camera_api_reference:

API Reference
=============

.. contents::
   :local:
   :class: this-will-duplicate-information-and-it-is-still-useful-here

> 

Reviewer: maybe some words here?

Camera
------

.. doxygengroup:: camera_main
   :project: lib_camera
   :members:

Sensors
-------

.. doxygengroup:: camera_sensors
   :project: lib_camera
   :members:

MIPI
----

.. doxygengroup:: camera_mipi
   :project: lib_camera

>    :members:

Reviewer: Do we really need to expose this here? it's a mipi conf struct and an init api, both used internally in camera_main

ISP
---

.. doxygengroup:: camera_isp_cfg
   :project: lib_camera
   :members:

.. doxygengroup:: camera_isp_api
   :project: lib_camera
   :members:

.. doxygengroup:: camera_isp_conv
   :project: lib_camera
   :members:

Utils
------

.. doxygengroup:: camera_utils
   :project: lib_camera

>    :members:

Reviewer: I would remove this as it's only there for internal use

|newpage|

.. _lib_camera_api_reference:

API Reference
=============

This section contains the API reference for the Camera library.

.. contents::
   :local:
   :class: this-will-duplicate-information-and-it-is-still-useful-here

Camera
------

The Camera module serves as the entry point for the camera library.

It defines global macros for main camera configuration and provides functions to configure and initialize the camera thread.

.. doxygengroup:: camera_main
   :project: lib_camera
   :members:

ISP
---

The ISP module offers an interface for configuring and controlling the camera's image processing pipeline.
It provides functions to set the output format, resolution, and various image processing parameters.
This module includes both configuration and conversion functions essential for adapting the camera output to different requirements.

.. doxygengroup:: camera_isp_cfg
   :project: lib_camera
   :members:

.. doxygengroup:: camera_isp_api
   :project: lib_camera
   :members:

> .. doxygengroup:: camera_isp_conv

Reviewer 2: Its not clear from reading this section what sort of value the `GET_[RGBYUV]` macros are for, possibly more words would help

   :project: lib_camera
   :members:

Sensors
-------

> This module provides a high-level class fort he camera sensor. 

Reviewer 2: ```suggestion
This module provides a high-level class for the camera sensor. 
```
